From 4c6df864be62d8b6dea7889b79da78a483682991 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Fri, 30 Oct 2020 09:50:53 +0100
Subject: [PATCH 3/3] ARM: dts: stm32: Add PicoITX DTO for DH 626-100 LVDS
 adapter with CH101OLHLWH display

Add DT overlay for the PicoITX board for DH 626-100 adapter card with
Chefree CH101OLHLWH-002 LVDS display attached to it. This is used to
form a DHMI system.

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@denx.de>
---
 arch/arm/boot/dts/Makefile                    |   1 +
 ...picoitx-overlay-626-100-x2-ch101olhlwh.dts | 116 ++++++++++++++++++
 2 files changed, 117 insertions(+)
 create mode 100644 arch/arm/boot/dts/stm32mp157c-dhcom-picoitx-overlay-626-100-x2-ch101olhlwh.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 5bce2e0d66c9..03c95e50aa77 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1011,6 +1011,7 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157c-dhcom-pdk2-overlay-560-200-x12.dtbo \
 	stm32mp157c-dhcom-pdk2-overlay-638-100-x12-rpi7inch.dtbo \
 	stm32mp157c-dhcom-picoitx.dtb \
+	stm32mp157c-dhcom-picoitx-overlay-626-100-x2-ch101olhlwh.dtbo \
 	stm32mp157c-dk2.dtb \
 	stm32mp157c-ed1.dtb \
 	stm32mp157c-ev1.dtb \
diff --git a/arch/arm/boot/dts/stm32mp157c-dhcom-picoitx-overlay-626-100-x2-ch101olhlwh.dts b/arch/arm/boot/dts/stm32mp157c-dhcom-picoitx-overlay-626-100-x2-ch101olhlwh.dts
new file mode 100644
index 000000000000..52541b669b87
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp157c-dhcom-picoitx-overlay-626-100-x2-ch101olhlwh.dts
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2020 Marek Vasut <marex@denx.de>
+ */
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/pwm/pwm.h>
+
+/dts-v1/;
+/plugin/;
+
+&{/} {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	display_bl: display-bl {
+		compatible = "pwm-backlight";
+		pwms = <&pwm2 3 5000000 0>;
+		brightness-levels = <0 16 22 30 40 55 75 102 138 188 255>;
+		default-brightness-level = <8>;
+		enable-gpios = <&gpioi 0 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+
+	lvds-encoder {
+		compatible = "onsemi,fin3385", "lvds-encoder";
+		pixelclk-active = <0>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+
+				lvds_bridge_in: endpoint {
+					remote-endpoint = <&ltdc_ep0_out>;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+
+				lvds_bridge_out: endpoint {
+					remote-endpoint = <&lvds_panel_in>;
+				};
+			};
+		};
+	};
+
+	panel {
+		compatible = "chefree,ch101olhlwh-002";
+		backlight = <&display_bl>;
+
+		port {
+			lvds_panel_in: endpoint {
+				remote-endpoint = <&lvds_bridge_out>;
+			};
+		};
+	};
+};
+
+&timers2 {
+	/* spare dmas for other usage (un-delete to enable pwm capture) */
+	dmas;
+	dma-names;
+	status = "okay";
+
+	pwm2: pwm {
+		#pwm-cells = <3>;
+		pinctrl-0 = <&pwm2_pins_a>;
+		pinctrl-names = "default";
+		status = "okay";
+	};
+
+	timer@1 {
+		status = "okay";
+	};
+};
+
+&ltdc {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&ltdc_pins_b>;
+	pinctrl-1 = <&ltdc_sleep_pins_b>;
+	status = "okay";
+
+	port {
+		ltdc_ep0_out: endpoint@0 {
+			reg = <0>;
+			remote-endpoint = <&lvds_bridge_in>;
+		};
+	};
+};
+
+&i2c5 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	ili251x@41 {
+		compatible = "ilitek,ili251x";
+		reg = <0x41>;
+		interrupt-parent = <&gpioi>;
+		interrupts = <2 IRQ_TYPE_EDGE_FALLING>;
+		reset-gpios = <&gpiod 6 GPIO_ACTIVE_LOW>;
+		touchscreen-size-x = <16384>;
+		touchscreen-size-y = <9600>;
+		touchscreen-inverted-x;
+		touchscreen-inverted-y;
+	};
+
+	eeprom@50 {
+		compatible = "atmel,24c04";
+		reg = <0x50>;
+		pagesize = <16>;
+	};
+};
-- 
2.28.0

